core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/PathImpl.kt

```diff
@@ -72,6 +72,21 @@ data class PathImpl(
         throw RuntimeException("The given path offset is larger than the path length")
     }
 
+    override fun getTrackSections(): List<TrackLocation> {
```

Comment:
This implementation doesn't handle every case: the first chunk may not be the first on its track, and the path may cover only parts of a chunk.

*Ideally* we'd handle projections in the same way as the other `Path` data. I get that it's not that easy here because we need extra electrification data, but maybe we could pass the mapping as parameters in the getters? (I don't have a precise plan)